api/client/builder/types.go

```diff
@@ -0,0 +1,549 @@
+	GasUsed          Uint64String `json:"gas_used,omitempty"`
+	Timestamp        Uint64String `json:"timestamp,omitempty"`
+	ExtraData        hexSlice     `json:"extra_data,omitempty"`
+	BaseFeePerGas    Uint64String `json:"base_fee_per_gas,omitempty"`
+	BlockHash        hexSlice     `json:"block_hash,omitempty"`
+	TransactionsRoot hexSlice     `json:"transactions_root,omitempty"`
+	*eth.ExecutionPayloadHeader
+}
+
+func (h *ExecutionPayloadHeader) MarshalJSON() ([]byte, error) {
+	// TODO check this encoding and confirm bounds of values
+	bfpg, err := strconv.ParseUint(string(h.ExecutionPayloadHeader.BaseFeePerGas), 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	type MarshalCaller ExecutionPayloadHeader
+	return json.Marshal(&MarshalCaller{
+		ParentHash:       h.ExecutionPayloadHeader.ParentHash,
+		FeeRecipient:     h.ExecutionPayloadHeader.FeeRecipient,
+		StateRoot:        h.ExecutionPayloadHeader.StateRoot,
+		ReceiptsRoot:     h.ExecutionPayloadHeader.ReceiptsRoot,
+		LogsBloom:        h.ExecutionPayloadHeader.LogsBloom,
+		PrevRandao:       h.ExecutionPayloadHeader.PrevRandao,
+		BlockNumber:      Uint64String(h.ExecutionPayloadHeader.BlockNumber),
+		GasLimit:         Uint64String(h.ExecutionPayloadHeader.GasLimit),
+		GasUsed:          Uint64String(h.ExecutionPayloadHeader.GasUsed),
+		Timestamp:        Uint64String(h.ExecutionPayloadHeader.Timestamp),
+		ExtraData:        h.ExecutionPayloadHeader.ExtraData,
+		BaseFeePerGas:    Uint64String(bfpg),
+		BlockHash:        h.ExecutionPayloadHeader.BlockHash,
+		TransactionsRoot: h.ExecutionPayloadHeader.TransactionsRoot,
+	})
+}
+
+type ExecPayloadResponse struct {
+	Version string           `json:"version,omitempty"`
+	Data    ExecutionPayload `json:"data,omitempty"`
+}
+
+type ExecutionPayload struct {
+	ParentHash    hexSlice     `json:"parent_hash,omitempty"`
+	FeeRecipient  hexSlice     `json:"fee_recipient,omitempty"`
+	StateRoot     hexSlice     `json:"state_root,omitempty"`
+	ReceiptsRoot  hexSlice     `json:"receipts_root,omitempty"`
+	LogsBloom     hexSlice     `json:"logs_bloom,omitempty"`
+	PrevRandao    hexSlice     `json:"prev_randao,omitempty"`
+	BlockNumber   Uint64String `json:"block_number,omitempty"`
+	GasLimit      Uint64String `json:"gas_limit,omitempty"`
+	GasUsed       Uint64String `json:"gas_used,omitempty"`
+	Timestamp     Uint64String `json:"timestamp,omitempty"`
+	ExtraData     hexSlice     `json:"extra_data,omitempty"`
+	BaseFeePerGas Uint64String `json:"base_fee_per_gas,omitempty"`
+	BlockHash     hexSlice     `json:"block_hash,omitempty"`
+	Transactions  []hexSlice   `json:"transactions,omitempty"`
+}
+
+func (r *ExecPayloadResponse) ToProto() (*v1.ExecutionPayload, error) {
+	return r.Data.ToProto()
+}
+
+func (p *ExecutionPayload) ToProto() (*v1.ExecutionPayload, error) {
+	// TODO: it looks like BaseFeePerGas should probably be a uint
+	baseFeeHack := []byte(strconv.FormatUint(uint64(p.BaseFeePerGas), 10))
```

Comment:
Same feed back as above, this should be big int

Reply:
fixed